### DISCUSSION TITLE
What happens if triggers are not stopped prior to deployment?

### DISCUSSION THREAD

**Discussion by dfindlay** [Azure-Player/azure.datafactory.tools#176] [Q&A] [answered]
Hi,
I am interested in what the impact would be if we did **not** stop the triggers prior to deployment. Would the deployment fail, or would there be any other undesired consequences?

I currently have our CI/CD deployment set up to stop all triggers prior to deployment and restart them afterwards (although using a custom Python script rather than via your options, as we only want to restart triggers that were already active prior to the deployment). However, we have hundreds of triggers and so this actually adds 20-30 minutes to our deployment time. It also means that some scheduled tasks could be missed due to the deployment window. It would therefore be good if we could just deploy without stopping the triggers, but I am not sure what the consequences would be.

@NowinskiK, do you have any experience with deploying without first stopping triggers, and can you advise what you saw happen in that case? And is the behaviour different for event-based vs scheduled triggers? Any advice you could give me would be much appreciated.

**Comment by NowinskiK** [comment, marked as answer]
Simply, you can't deploy the trigger if it's active (enabled/running). You'll get an error.
If you have many objects in ADF you can try to split them logically and deploy them in groups.
